### PR TITLE
 🐛 Correctly identify if patch call was made on status

### PR DESCRIPTION
**Pull request by acumino**
Fixes https://github.com/kubernetes-sigs/controller-runtime/issues/2503

### CODE REVIEW

File: pkg/client/fake/client.go [hunk 1]
```diff
@@ -361,7 +361,7 @@ func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Ob
 	isStatus := false
 	// We apply patches using a client-go reaction that ends up calling the trackers Update. As we can't change
 	// that reaction, we use the callstack to figure out if this originated from the status client.
-	if bytes.Contains(debug.Stack(), []byte("sigs.k8s.io/controller-runtime/pkg/client/fake.(*fakeSubResourceClient).Patch")) {
+	if bytes.Contains(debug.Stack(), []byte("sigs.k8s.io/controller-runtime/pkg/client/fake.(*fakeSubResourceClient).statusPatch")) {
```
Comment (alvaroaleman) [inline pkg/client/fake/client.go:364]: Please provide a testcase that starts passing with this change

Comment (acumino) [reply pkg/client/fake/client.go:364]: There are multiple tests for the status subresource, which passes after the changes made in this PR also, I added a test to verify that subresource other than status can also be patched. PTAL

File: pkg/client/fake/client_test.go [hunk 2]
```diff
@@ -1590,6 +1591,28 @@ var _ = Describe("Fake client", func() {
 		Expect(objOriginal.Status.Phase).ToNot(Equal(actual.Status.Phase))
 	})
 
+	It("should be able to change typed objects that have a scale subresource on patch", func() {
```
Comment (alvaroaleman) [inline pkg/client/fake/client_test.go:1594]: I have to say that I am amazed this works at all. It ends up patching the original object, not some subresource which happens to work if the original object has a `.spec.replicas` field, which doesn't have to be the case. I'll accept this patch, but I will also note that the `scale` subresource isn't really supported in the fake client and just coincidentally works under some specific coincidences

**Review by alvaroaleman — APPROVED** [review]

File: pkg/client/fake/client_test.go (hunk 2 above)
Comment (acumino) [reply pkg/client/fake/client_test.go:1594]: > It ends up patching the original object, not some subresource which happens to work if the original object has a .spec.replicas field, which doesn't have to be the case. I'll accept this patch, but I will also note that the scale subresource isn't really supported in the fake client and just coincidentally works under some specific coincidences

This is true for any subresource other than status.

**Comment by acumino** [comment]
@alvaroaleman Can we cherry-pick this PR in v0.16?

**Pull request merged**